### PR TITLE
Fix link with emoji

### PR DESCRIPTION
**Pull request by ysakasin**
Fix issue:
https://github.com/gitlabhq/gitlabhq/issues/9413
https://gitlab.com/gitlab-org/gitlab-ce/issues/2349

When exec `link_to_gfm(':book:Book', '/foo')`, The value of `gfm_body` is:

```
<img class="emoji" title=":book:" alt=":book:" src="path/to/emoji.png" height="20" width="20" align="absmiddle">Book
```

The result of `Nokogiri::XML::DocumentFragment.parse(gfm_body)` is:

```
<img class="emoji" title=":book:" alt=":book:" src="path/to/emoji.png" height="20" width="20" align="absmiddle">Book</img>
```

because the img tag has not an end tag and it is not XML. 
So `Nokogiri::HTML::DocumentFragment.parse(gfm_body)` is returned the expected value which is `img` tag and `Book`.

### CODE REVIEW

**Comment by stanhu** [comment]
Thanks for looking into this. I am confused why there is `...` in the `img` tag in the first place. Also, why is there no closing `</img>` tag?

**Comment by NKMR6194** [comment]
Sorry, `...` means omission. I has changed the discription.

Emoji filter return `img` tag with end slash ([this line](https://github.com/gitlabhq/gitlabhq/blob/54452412f765d9e6e6166e105db9adbc7553aec2/lib/gitlab/markdown/emoji_filter.rb#L42)) but `HTML::Pipeline` maybe remove that slash.

`</img>` tag is not usually used but I think using `</img>` is nice.

**Comment by stanhu** [comment]
Hmm, sounds like `HTTML::Pipeline` should not be removing that slash?

/cc: @tsigo

**Comment by NKMR6194** [comment]
I searched about `img` tag on HTML5.
I think `HTTML::Pipeline` should be removing that slash on HTML5.

* `img` tag is [void element](http://www.w3.org/TR/html5/syntax.html#void-elements). It can't have any contents and end tag.
*  `` is syntax sugar of ``. (ref: [start tags](http://www.w3.org/TR/html5/syntax.html#start-tags))

**Comment by stanhu** [comment]
Oh right, that makes sense.

**Comment by stanhu** [comment]
Looks like the build is failing:

```
  1) GitlabMarkdownHelper#link_to_gfm should replase commit message with emoji to link
     Failure/Error: expect(act).
     NameError:
       undefined local variable or method `act' for #<RSpec::ExampleGroups::GitlabMarkdownHelper::LinkToGfm:0x007fb8d5006488>
     # ./spec/helpers/gitlab_markdown_helper_spec.rb:112:in `block (3 levels) in <top (required)>'
```

File: spec/helpers/gitlab_markdown_helper_spec.rb [hunk 1]
```diff
@@ -106,6 +106,12 @@
       act = link_to_gfm(text, '/foo')
       expect(act).to eq %Q(<a href="/foo">#{issues[0].to_reference}</a>)
     end
+
+    it 'should replase commit message with emoji to link' do
```
Comment (stanhu) [inline spec/helpers/gitlab_markdown_helper_spec.rb]: Misspelling here: replase -> replace

File: spec/helpers/gitlab_markdown_helper_spec.rb [hunk 2]
```diff
@@ -106,6 +106,12 @@
...
+      actual = link_to_gfm(':book:Book', '/foo')
+      expect(act).
```
Comment (stanhu) [inline spec/helpers/gitlab_markdown_helper_spec.rb]: I think you meant `expect(actual)`

**Comment by stanhu** [comment]
Thanks. Looks good. Can you squash this into one commit?

**Comment by NKMR6194** [comment]
Thanks reviewing. I have squashed commits. Is it OK?

**Comment by stanhu** [comment]
Looks good to me. @tsigo, is there any reason that parsing as HTML would cause issues? I notice some of the filters explicitly say `Returns a Nokogiri::XML::DocumentFragment`.

**Comment by tsigo** [comment]
Hating myself right now for how simple this fix was :+1: 

LGTM.

**Comment by stanhu** [comment]
Thanks, @NKMR6194!

**Comment by NKMR6194** [comment]
:smile:

**Pull request merged**